detox/src/configuration/composeAppsConfig.js

```diff
@@ -16,11 +18,14 @@ const CLI_PARSER_OPTIONS = {
  * @param {Detox.DetoxDeviceConfig} opts.deviceConfig
  * @param {Detox.DetoxConfiguration} opts.localConfig
  * @param {*} opts.cliConfig
+ * @param {Boolean} opts.isCloudSession
  * @returns {Record<string, Detox.DetoxAppConfig>}
  */
 function composeAppsConfig(opts) {
   const appsConfig = composeAppsConfigFromAliased(opts);
-  overrideAppLaunchArgs(appsConfig, opts.cliConfig);
+  if (!opts.isCloudSession) {
+    overrideAppLaunchArgs(appsConfig, opts.cliConfig);
```

Comment:
**[NOT A MUST HAVE]**
We should have ability in detox to prevent CLI override for particular device types. Implemented it for cloud device for now.

Reply:
For eg: --device-name can be used to override whole device object, but certain caps like `os`, `deviceName` are mandatory for cloud session